components/AlertsDashboard.vue

```diff
@@ -1973,8 +2090,454 @@ const calculateLineStringCentroid = (coordinates: number[][]) => {
+      const pixelBbox: [[number, number], [number, number]] = [
+        [Math.min(bbox[0].x, bbox[1].x), Math.min(bbox[0].y, bbox[1].y)],
+        [Math.max(bbox[0].x, bbox[1].x), Math.max(bbox[0].y, bbox[1].y)],
+      ];
+
+      selectFeaturesInBoundingBox(pixelBbox);
+    }
+
+    map.value.dragPan.enable();
+
+    start = null;
+    current = null;
+  };
+
+  canvas.addEventListener("mousedown", mouseDown, true);
+
+  (
+    map.value as mapboxgl.Map & { _boundingBoxCleanup?: () => void }
+  )._boundingBoxCleanup = () => {
+    canvas.removeEventListener("mousedown", mouseDown, true);
+    document.removeEventListener("mousemove", onMouseMove);
+    document.removeEventListener("keydown", onKeyDown);
+    document.removeEventListener("mouseup", onMouseUp);
+    if (box) {
+      box.parentNode?.removeChild(box);
+      box = null;
+    }
+    map.value.dragPan.enable();
+  };
+};
+
+/**
+ * Removes bounding box drawing handlers
+ */
+const removeBoundingBoxHandlers = () => {
+  if (!map.value) return;
+
+  // Call cleanup function if it exists
+  const cleanup = (
+    map.value as mapboxgl.Map & { _boundingBoxCleanup?: () => void }
+  )._boundingBoxCleanup;
+  if (cleanup) {
+    cleanup();
+    delete (map.value as mapboxgl.Map & { _boundingBoxCleanup?: () => void })
+      ._boundingBoxCleanup;
+  }
+
+  map.value.dragRotate.enable();
+};
+
+/**
+ * Selects features within the bounding box using pixel coordinates
+ * Uses the same robust layer checking as multi-select
+ */
+const selectFeaturesInBoundingBox = (
+  bbox: [[number, number], [number, number]],
+) => {
+  if (!map.value) return;
+
+  // Include centroids layers as they contain the alertID property
+  const alertLayers = [
+    "most-recent-alerts-polygon",
+    "most-recent-alerts-linestring",
+    "most-recent-alerts-point",
+    "most-recent-alerts-symbol",
+    "most-recent-alerts-centroids",
+    "previous-alerts-polygon",
+    "previous-alerts-linestring",
+    "previous-alerts-point",
+    "previous-alerts-symbol",
+    "previous-alerts-centroids",
+  ];
+
+  const mapeoLayers = ["mapeo-data"];
+
+  const allFeatures: Array<{
+    properties?: {
+      alertID?: string;
+      id?: string;
+      cluster?: boolean;
+      cluster_id?: number;
+    };
+    layer?: { id?: string };
+  }> = [];
+
+  alertLayers.forEach((layerId) => {
+    try {
+      if (map.value.getLayer(layerId)) {
+        const features = map.value.queryRenderedFeatures(bbox, {
+          layers: [layerId],
+        });
+        const validFeatures = features.filter(
+          (f) =>
+            !f.properties?.cluster &&
+            f.properties?.cluster_id === undefined &&
+            !f.layer?.id?.includes("clusters") &&
+            !f.layer?.id?.includes("cluster-count"),
+        );
+        allFeatures.push(...validFeatures);
+        console.log(`Layer ${layerId}: found ${validFeatures.length} features`);
```

Comment:
I don't know if we've established team norms for logging, but at the very least I suggest replacing these `console.log` with a specific log level (likely `console.debug` in most cases)